src/libraries/System.Net.WebProxy/src/System/Net/WebProxy.cs

```diff
@@ -94,51 +94,51 @@ public Uri GetProxy(Uri destination)
             return IsBypassed(destination) ? destination : Address;
         }
 
-        private static Uri CreateProxyUri(string address) =>
+        private static Uri? CreateProxyUri(string? address) =>
             address == null ? null :
             address.IndexOf("://", StringComparison.Ordinal) == -1 ? new Uri("http://" + address) :
             new Uri(address);
 
-        private void UpdateRegExList(bool canThrow)
+        private void UpdateRegexList(bool canThrow)
         {
-            Regex[] regExBypassList = null;
-            ArrayList bypassList = _bypassList;
+            Regex[]? regexBypassList = null;
+            ArrayList? bypassList = _bypassList;
             try
             {
                 if (bypassList != null && bypassList.Count > 0)
                 {
-                    regExBypassList = new Regex[bypassList.Count];
+                    regexBypassList = new Regex[bypassList.Count];
                     for (int i = 0; i < bypassList.Count; i++)
                     {
-                        regExBypassList[i] = new Regex((string)bypassList[i], RegexOptions.IgnoreCase | RegexOptions.CultureInvariant);
+                        regexBypassList[i] = new Regex((string)bypassList[i]!, RegexOptions.IgnoreCase | RegexOptions.CultureInvariant);
```

Comment:
What is ensuring `bypassList[i]` is non-null?

Reply:
It's constructed from a `string[]`:
https://github.com/dotnet/runtime/pull/32612/files#diff-9818c815d0205e946ffc3615ef703438R33
so as long as that annotation wasn't violated, these won't be null.

Reply:
What about when I set the public `public string[] BypassList` property?

Reply:
oh - that is a `string[]`, not a `string?[]`.

Reply:
Right.

Reply:
But I could have added `null` to `public ArrayList BypassArrayList`. I'm surprised we expose that...

https://github.com/dotnet/runtime/pull/32612/files#diff-9818c815d0205e946ffc3615ef703438R77

Reply:
> But I could have added null to public ArrayList BypassArrayList. I'm surprised we expose that...

I didn't notice we expose that.  In that case, yes, if someone adds null, this could throw an ArgumentNullException... which seems fairly appropriate, if delayed.